test/src/tests.js

```diff
@@ -139,4 +174,577 @@ describe('Bing Ads Event Forwarder', function() {
+                ea: 'pageLoad',
+                ec: 'This is my name!',
+                el: 'Test Page Event',
+                ev: 10,
+            });
+
+            done();
+        });
+
+        it('should create a Consent State Default of Granted if consent mappings and settings are undefined', function(done) {
+            mParticle.forwarder.init(
+                {
+                    tagId: 'tagId',
+                },
+                reportService.cb,
+                false // Disable testMode so we can test init
+            );
+
+            var expectedInitialConsentPayload = [
+                'consent',
+                'default',
+                { ad_storage: 'granted' },
+            ];
+
+            // UETQ queues up events as array elements and then parses them internally.
+            // The first 3 elements of this array will be the consent payload
+
+            window.uetq.length.should.eql(3);
+            window.uetq[0].should.equal('consent');
+            window.uetq[1].should.equal('default');
+            window.uetq[2].should.eql(expectedInitialConsentPayload[2]);
+
+            var obj = {
+                EventDataType: MessageType.PageEvent,
+                EventName: 'Test Page Event',
+                CustomFlags: {
+                    'Bing.EventValue': 10,
+                },
+                ConsentState: {
+                    getGDPRConsentState: function() {
+                        return {
+                            marketing_consent: {
+                                Consented: true,
+                                Timestamp: 1557935884509,
+                                ConsentDocument: 'Marketing_Consent',
+                                Location: 'This is fake',
+                                HardwareId: '123456',
+                            },
+                        };
+                    },
+
+                    getCCPAConsentState: function() {
+                        return {
+                            data_sale_opt_out: {
+                                Consented: false,
+                                Timestamp: Date.now(),
+                                Document: 'some_consent',
+                            },
+                        };
+                    },
+                },
+            };
+
+            mParticle.forwarder.process(obj);
+
+            // UETQ should now have 7 elements
+            // The first 3 elements of this array will be the initial consent payload
+            // The 4th element will be the event payload
+
+            window.uetq.length.should.eql(4);
+            window.uetq[3].should.eql({
+                ea: 'pageLoad',
+                ec: 'This is my name!',
+                el: 'Test Page Event',
+                ev: 10,
+            });
+
+            done();
+        });
+
+        it('should construct Consent State Payloads if consent mappings is undefined but settings defaults are defined', function(done) {
```

Comment:
maybe this should be "should construct default COnsent Payloads if..........but should NOT construct updates"